_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-mailchimp.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * Mailchimp: Get Mailchimp Status.
+ * API to determine if current site has linked Mailchimp account and mailing list selected.
+ * This API is meant to be used in Jetpack and on WPCOM.
+ *
+ * @since 7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Mailchimp extends WP_REST_Controller {
```

Comment:
My personal preference:
I really like keeping files touching a functionality together.
Is there a way we could keep it with other files touching that block?

Unless there is a pattern to do it this way.

Don't mind me really - feel free to ignore that comment.
Just voicing an opinion

Reply:
> Unless there is a pattern to do it this way.

There is, but it's not really documented here in the plugin; all documentation about this is currently internal. p3hLNG-12Y-p2 is a good resource to find out more, as well as gh-api-docs-internal.

We currently keep all v2 WP REST API endpoints here, so they can easily be synchronized with WordPress.com:
https://github.com/Automattic/jetpack/tree/2655ec90865057cbec48a6622ff5b09891df5911/_inc/lib/core-api

Reply:
If its good with you - I'm happy :)